From: Xaver Hugl <xaver.hugl@kde.org>
Date: Tue, 28 Jul 2026 17:35:52 +0200
Subject: [PATCH] opengl/eglcontext: optimize makeCurrent if the same context
 is already current

I'm not sure if the driver will optimize that out anyways, but this ensures we
don't need to worry about it.

Upstream-Status: Backport [https://invent.kde.org/plasma/kwin/-/commit/ec1db4366c5a159013dd826ebb53f3ba210ead18]
---
 src/opengl/eglcontext.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/opengl/eglcontext.cpp b/src/opengl/eglcontext.cpp
--- a/src/opengl/eglcontext.cpp
+++ b/src/opengl/eglcontext.cpp
@@ -127,6 +127,9 @@ bool EglContext::makeCurrent()
     if (QOpenGLContext *context = QOpenGLContext::currentContext()) {
         // Workaround to tell Qt that no QOpenGLContext is current
         context->doneCurrent();
+    } else if (s_currentContext == this) {
+        // nothing to do
+        return true;
     }
     const bool ret = eglMakeCurrent(m_display->handle(), EGL_NO_SURFACE, EGL_NO_SURFACE, m_handle) == EGL_TRUE;
     if (ret) {
